cmd/key_rotate.go

```diff
@@ -0,0 +1,162 @@
+		s, err := wallets.InitialiseStore(rf.Home)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't initialise wallets store: %w", err)
+		}
+
+		return wallet.RotateKey(s, req)
+	}
+
+	return BuildCmdRotateKey(w, h, rf)
+}
+
+func BuildCmdRotateKey(w io.Writer, handler RotateKeyHandler, rf *RootFlags) *cobra.Command {
+	f := RotateKeyFlags{}
+
+	cmd := &cobra.Command{
+		Use:     "rotate",
+		Short:   "Get signed key rotation transaction",
+		Long:    rotateKeyLong,
+		Example: rotateKeyExample,
+		RunE: func(_ *cobra.Command, args []string) error {
+			if aLen := len(args); aLen == 0 {
+				return flags.ArgMustBeSpecifiedError("public-key")
+			} else if aLen > 1 {
+				return flags.TooManyArgsError("public-key")
+			}
+			f.NewPublicKey = args[0]
+
+			req, err := f.Validate()
+			if err != nil {
+				return err
+			}
+
+			resp, err := handler(req)
+			if err != nil {
+				return err
+			}
+
+			switch rf.Output {
+			case flags.InteractiveOutput:
+				PrintRotateKeyResponse(w, resp)
+			case flags.JSONOutput:
+				return printer.FprintJSON(w, resp)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&f.Wallet,
+		"wallet", "w",
+		"",
+		"Wallet holding the master key and new public key",
+	)
+	cmd.Flags().StringVarP(&f.PassphraseFile,
+		"passphrase-file", "p",
+		"",
+		"Path to the file containing the wallet's passphrase",
+	)
+	cmd.Flags().Uint32VarP(&f.TXBlockHeight,
+		"tx-height", "xh",
+		0,
+		"Block height of which the transaction will be applied at",
+	)
+	cmd.Flags().Uint32VarP(&f.TargetBlockHeight,
+		"target-height", "th",
+		0,
+		"Block height from which the new public key will be used",
+	)
+
+	return cmd
+}
+
+type RotateKeyFlags struct {
+	Wallet            string
+	PassphraseFile    string
+	NewPublicKey      string
+	TXBlockHeight     uint32
+	TargetBlockHeight uint32
+}
+
+func (f *RotateKeyFlags) Validate() (*wallet.RotateKeyRequest, error) {
+	req := &wallet.RotateKeyRequest{
+		NewPublicKey: f.NewPublicKey,
+	}
+
+	if f.TargetBlockHeight == 0 {
+		return nil, flags.FlagMustBeSpecifiedError("target-height")
+	}
+	req.TargetBlockHeight = f.TargetBlockHeight
+
+	if f.TXBlockHeight == 0 {
+		return nil, flags.FlagMustBeSpecifiedError("tx-height")
+	}
+	req.TXBlockHeight = f.TXBlockHeight
+
+	if req.TargetBlockHeight <= req.TXBlockHeight {
+		return nil, errTargetLessThenTxHeigh
+	}
+
+	if len(f.Wallet) == 0 {
+		return nil, flags.FlagMustBeSpecifiedError("wallet")
+	}
+	req.Wallet = f.Wallet
+
+	passphrase, err := flags.GetPassphrase(f.PassphraseFile)
+	if err != nil {
+		return nil, err
+	}
+	req.Passphrase = passphrase
+
+	return req, nil
+}
+
+func PrintRotateKeyResponse(w io.Writer, req *wallet.RotateKeyResponse) {
+	p := printer.NewInteractivePrinter(w)
+	p.CheckMark().SuccessText("Key rotation succeeded").NextSection()
+	p.Text("Base64 encoded transaction:").NextLine()
+	p.Text(req.Base64Transaction).NextLine()
+	p.Text("New public key:").NextLine()
+	p.Text(req.NewPublicKey).NextLine()
```

Comment:
nit-pick: I don't think we need to display it as it's already set in the flags.

Reply:
Yeah.. I was just trying to be fool proof.. maybe too much haha